core/src/util/pbfParser.cpp

```diff
@@ -124,7 +130,17 @@ void PbfParser::extractFeature(ParserContext& _ctx, protobuf::message& _featureI
                 break;
         }
     }
-    _out.props = std::move(_ctx.properties);
+
+    std::vector<Properties::Item> properties;
+    properties.reserve(_ctx.featureTags.size());
+
+    for (int tagKey : _ctx.orderedKeys) {
+        int tagValue = _ctx.featureTags[tagKey];
+        if (tagValue >= 0) {
```

Comment:
This iterates through all keys of layer in the order in which Property(s) need to be added, so that the Properties is sorted correctly.
featureTags were initialized with -1 in the beginning of extractFeature to be able to check whether the value got assigned.